app/javascript/src/components/form/Derivatives.svelte

```diff
@@ -0,0 +1,84 @@
+<script>
+  import Tags from './Tags.svelte';
+  import { onMount } from 'svelte';
+
+  let showDerivative = false;
+  let maxCodes = 5;
+  export let currSources = [];
+
+  onMount(() => {
+    if (currSources.length) showDerivative = true;
+  });
+
+  async function handleAutoCompleteRequest(value) {
+    if (!value) return [];
+
+    const timeout = 8000;
+    const timeoutController = new AbortController();
+    const timeoutID = setTimeout(() => timeoutController.abort(), timeout);
+
+    const response = await fetch(`/code/${value}`, {
```

Comment:
There is a `fetch-rails.js` import you could use for fetch requests instead. Could be used like

```
new FetchRails(`/code/${value}`).get()
  .then(data => {
      const json = JSON.parse(text)
      return json.map(post => postToResult(post))
  })
  .error(error => {
    throw new Error(`${response.status} ${response.statusText}`)
  })
``

Although you wouldn't have the timeout you created above, but perhaps it would make sense to add the timeout to the `fetch-rails.js` so it applies to everything.
I don't mind your solution here either, it seems nice enough. So feel free to decide if you want to do this or not.

Reply:
I was unaware that `fetch-rails.js` existed. I'll probably add a timeout feature to it and DRY the code.